src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -1092,15 +1124,18 @@ private string ResolvePSPath(string path)
             {
                 ProviderInfo provider = null;
                 string resolvedPath = null;
-                if (WildcardPattern.ContainsWildcardCharacters(path))
+
+                // Try literal path resolution if it is set to run first
+                if (_commandResolutionOptions.HasFlag(SearchResolutionOptions.ResolveLiteralThenPathPatterns))
+                {
+                    resolvedPath = GetNextLiteralPathThatExists(path, out provider);
+                }
+
+                if (WildcardPattern.ContainsWildcardCharacters(path) &&
```

Comment:
Here we check for `resolvedPath == null` instead of just returning as above.  Please add comments as to why.

Reply:
Added comment above